GameServerLib/Content/ZipPackage.cs

```diff
@@ -0,0 +1,250 @@
+            {
+                _logger.Debug($"Package: {PackageName} does not contain file for {itemName}.");
+                return null;
+            }
+
+            var filePath = $"{PackageName}/{contentType}/{itemName}/{itemName}.json";
+
+            IContentFile toReturnContentFile;
+
+            using (var archive = ZipFile.OpenRead(PackagePath))
+            {
+                var file = archive.GetEntry(filePath);
+
+                if (file == null)
+                {
+                    return null;
+                }
+
+                var fileText = new StreamReader(file.Open(), Encoding.Default).ReadToEnd();
+                toReturnContentFile = JsonConvert.DeserializeObject<ContentFile>(fileText);
+            }
+
+            return toReturnContentFile;
+        }
+
+        public IMapSpawns GetMapSpawns(int mapId)
+        {
+            const string contentType = "Maps";
+            var mapName = $"Map{mapId}";
+            var filePath = $"{PackageName}/{contentType}/{mapName}/{mapName}.json";
+            var toReturnMapSpawns = new MapSpawns();
+
+            if (!_content.ContainsKey(contentType) || !_content[contentType].ContainsKey(mapName))
+            {
+                _logger.Debug($"Package: {PackageName} does not contain file for {mapName}.");
+                return null;
+            }
+
+            JToken mapSpawnInformation;
+
+            using (var archive = ZipFile.OpenRead(PackagePath))
+            {
+                var file = archive.GetEntry(filePath);
+
+                if (file == null)
+                {
+                    return null;
+                }
+
+                var fileText = new StreamReader(file.Open(), Encoding.Default).ReadToEnd();
+
+                try
+                {
+                    var mapData = JObject.Parse(fileText);
+
+                    mapSpawnInformation = mapData.SelectToken("spawns");
+                }
+                catch (JsonReaderException)
+                {
+                    return null;
+                }
+            }
+
+            foreach (JProperty teamSpawn in mapSpawnInformation)
+            {
+                var team = teamSpawn.Name;
+                var spawnsByPlayerCount = (JArray)teamSpawn.Value;
+                for (var i = 0; i < spawnsByPlayerCount.Count; i++)
+                {
+                    var playerSpawns = new PlayerSpawns((JArray)spawnsByPlayerCount[i]);
+                    toReturnMapSpawns.SetSpawns(team, playerSpawns, i);
+                }
+            }
+
+            return toReturnMapSpawns;
+        }
+
+        public INavGrid GetNavGrid(int mapId)
+        {
+            var navgridPath = $"{PackageName}/AIMesh/Map{mapId}/AIPath.aimesh_ngrid";
+
+            string fileText;
+
+            using (var archive = ZipFile.OpenRead(PackagePath))
+            {
+                var file = archive.GetEntry(navgridPath);
```

Comment:
I see these lines repeat many times. Consider doing something like:

In `PackageBase` add the get entry function:
`public Stream GetPackageEntry(string entryToken){}`
implement all these functions as `GetNavGrid` in `PackageBase`. Then you only need to change implement the entry function and you have all the others.